validator/client/iface/beacon_chain_client.go

```diff
@@ -4,9 +4,22 @@ import (
 	"context"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/validator"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 )
 
+var ErrNotSupported = errors.New("endpoint not supported")
+
+type ValidatorCount struct {
+	Status string
+	Count  uint64
+}
+
+type ValidatorCountProvider interface {
```

Comment:
@rkapka do you mean adding `GetValidatorCount` to `ValidatorClient` interface in `validator/client/iface/validator_client.go`?

We can do that but then we will need to implement `grpc_validator_client` method for this interface in `validator/client/grpc-api/grpc-validator-client.go` which I tried to avoid.